src/actors/symbolication.rs

```diff
@@ -882,21 +883,19 @@ pub struct SymbolicateStacktraces {
 }
 
 impl SymbolicationActor {
-    fn do_symbolicate(
-        &self,
+    async fn do_symbolicate(
+        self,
         request: SymbolicateStacktraces,
-    ) -> ResponseFuture<CompletedSymbolicationResponse, SymbolicationError> {
-        let result = self
-            .clone()
-            .do_symbolicate_impl(request)
-            .boxed_local()
-            .compat();
+    ) -> Result<CompletedSymbolicationResponse, SymbolicationError> {
+        let result = self.do_symbolicate_impl(request).boxed_local().compat();
 
-        Box::new(future_metrics!(
+        future_metrics!(
```

Comment:
Before we continue with this PR, can we first create a version of `future_metrics!`, or ideally a combinator, that works with std future? In the current way, the code is less readable than it was before.